cfme/tests/infrastructure/test_host_role_type.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+from utils import testgen
+from cfme.web_ui import Quadicon
+from cfme.web_ui import InfoBlock
+from cfme.fixtures import pytest_selenium as sel
+
+
+def pytest_generate_tests(metafunc):
+    argnames, argvalues, idlist = \
+        testgen.infra_providers(metafunc, required_fields=["ssh_credentials"])
+    testgen.parametrize(metafunc, argnames, argvalues, ids=idlist,
+                        scope="module")
+ROLES = ['Compute', 'Controller', 'BlockStorage', 'SwiftStorage',
+         'CephStorage']
+
+
+@pytest.mark.usefixtures("setup_provider_modscope")
+def test_host_role_type(provider):
+    provider.load_details()
+    sel.click(InfoBlock.element("Relationships", "Nodes"))
+    my_quads = list(Quadicon.all())
+    assert len(my_quads) > 0
+    result = False
+    for quad in my_quads:
+        role_name = str(quad).split(" ")[1].replace(')', '').replace('(', '')
+        if role_name in ROLES:
+            result = True
```

Comment:
* Don't you want to store all the types that were encountered during the browsing in some list so you can then check whether the roles that are detected by CFME indeed are those that are in openstack?
* You are going to this view, therefore you would expect that *ALL* of the hosts will be corresponding to this pattern, so I guess that instead of result = True, there should be soft_assert(role_name in ROLES, " ... ")